orttraining/orttraining/python/ort_trainer.py

```diff
@@ -691,6 +693,70 @@ def load_state_dict(self, state_dict, strict=False):
             session_state[name] = state_dict[name].numpy()
         self.session.load_state(session_state, strict)
 
+    def save_checkpoint(self, checkpoint_dir, checkpoint_prefix="ORT_checkpoint", checkpoint_state_dict=None):
```

Comment:
As per offline conversation, we would like to see checkpoint API being moved away from ORTTRainer, so that ORTModel.

For such, we are ok assuming that:
1) 'model' (ORTTrainer and ORTModel) has atributes world_rank, world_size and partition_optimizer
1.1) That means ORTModel needs to add a static partition_optimizer=False to match ORTTrainer
2) self and checkpoint_state_dict can be merged into a "obj" argument. To support both ORTTrainer and ORTModel,, we need:
2.1) Fort ORTTrainer, obj can be ORTTrainer or a dict which has an ORTTrainer inside. The name for this name is not really important, but can be assumed to be 'model', if preferred. Otherwise, just iterate over the dict looking for an instance of ORTTrainer and you get the name
2.2) For ORTModel, obj must be a dict with both ORTModel and torch.optim.Optimizer instances. 'model' and 'optimizer' can be assumed or discovered during dict iteration
2.3) Both models can persist additional elements in the dictionary
3) replace checkpoint_prefix by 'tag', like DeepSpeed API